### PR TITLE
Add schema documentation for the AWSControlPlane CRD

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/8804

This PR adds

- YAML file generation in unit tests
- property descriptions to the validation schema

### CODE REVIEW

File: docs/crd/infrastructure.giantswarm.io_awscontrolplane.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                  Configures which AWS availability zones to use by master nodes.
+                  We support either 1 or 3 availability zones.
+                items:
+                  description: |
+                    Identifier of an availability zone to use.
+                  type: string
+                type: array
+              instanceType:
+                description: |
+                  EC2 instane type to use for all master nodes.
```
Comment (anvddriesch) [inline docs/crd/infrastructure.giantswarm.io_awscontrolplane.yaml]:
```suggestion
                  EC2 instance type to use for all master nodes.
```

File: pkg/apis/infrastructure/v1alpha2/aws_control_plane_types.go [hunk 2]
```diff
@@ -35,17 +39,27 @@ spec:
...
+                  We support either 1 or 3 availability zones.
+                type: array
                 items:
+                  description: |
+                    Identifier of an availability zone to use.
                   type: string
-                type: array
               instanceType:
+                description: |
+                  EC2 instane type to use for all master nodes.
```
Comment (anvddriesch) [inline pkg/apis/infrastructure/v1alpha2/aws_control_plane_types.go]:
```suggestion
                  EC2 instance type to use for all master nodes.
```

**Comment by marians** [comment]
@anvddriesch Are you fine with this now?

**Review by anvddriesch — APPROVED** [review]
Looks good : )

**Pull request merged**